tests/be_in_matcher.go

```diff
@@ -0,0 +1,102 @@
+package tests
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/matchers"
+	"github.com/onsi/gomega/types"
+)
+
+//BeIn succeeds if actual is contained in the passed in elements.
+//BeIn() always uses Equal() to perform the match.
+//When the passed in elements are comprised of a single element that is either a Map or Array or Slice, BeIn() behaves
+//as the reverse of ContainElement() that operates with Equal() to perform the match.
+//    Expect(2).Should(BeIn([]int{1, 2}))
```

Comment:
That is a nice matcher, but you can express it already the other way round:

```golang
Expect(map[string]int{"foo": 1, "bar": 2})).To(HaveKey("foo"))
Expect(map[string]int{"foo": 1, "bar": 2})).To(HaveKeyWithValue("foo", 1))
Expect([]string{"a", "b"}).To(ContainElement("a"))
```

Also this is possible: 

```golang
Expect(myfunc()).To(Or(Equal("a"), Equal("b")))
```

Reply:
> That is a nice matcher, but you can express it already the other way round:
> 
> ```go
> Expect(map[string]int{"foo": 1, "bar": 2})).To(HaveKey("foo"))
> Expect(map[string]int{"foo": 1, "bar": 2})).To(HaveKeyWithValue("foo", 1))
> Expect([]string{"a", "b"}).To(ContainElement("a"))
> ```

Right, in those cases, the order is insignificant (besides if we wish to keep the convention of having `ACTUAL` be the computed item and `VALUE` be its desired value consistent across the tests and `ACTUAL` is not a Map/Array/Slice).

But in other cases, the order may be significant. For example, when using `Eventually`:
```golang
Eventually(fn).Should(BeIn(phases))
```

> 
> Also this is possible:
> 
> ```go
> Expect(myfunc()).To(Or(Equal("a"), Equal("b")))
> ```

This alternative also requires me to have the values I compare the returned value of `myfunc` with hard-coded and in my case, I want them to be provided as an argument to `waitForVMIPhase`

Reply:
A nice matcher.